pdcupdater/handlers/depchain/rpms.py

```diff
@@ -0,0 +1,249 @@
+
+        return True
+
+    def _yield_pdc_relationships(self, pdc, release_id):
+        for type in self.managed_types:
+            entries = pdc.get_paged(
+                pdc['release-component-relationships']._,
+                from_component_release=release_id,
+                type=type,
+            )
+            for entry in entries:
+                yield entry
+
+    def _yield_koji_relationships(self, pdc, tag):
+
+        release_id, release = tag2release(tag)
+        # TODO -- this tag <-> release agreement is going to break down with modularity.
+
+        pdcupdater.utils.ensure_release_exists(pdc, release_id, release)
+
+        builds = pdcupdater.services.koji_builds_in_tag(self.koji_url, tag)
+
+        for build in builds:
+            parent = {
+                'name': build['name'],
+                'release': {'release_id': release_id},
+            }
+
+            relationships = self.get_koji_relationships_from_build(
+                self.koji_url, build['build_id'])
+            relationships = list(relationships)
+
+            for relationship, child in relationships:
+                child = {'name': child, 'release': {'release_id': release_id}}
+                yield parent, relationship, child
+
+    def handle(self, pdc, msg):
+        tag = msg['msg']['tag']
+        release_id, release = tag2release(tag)
+
+        # TODO -- this tag <-> release agreement is going to break down with modularity.
+        pdcupdater.utils.ensure_release_exists(pdc, release_id, release)
+
+        name = msg['msg']['name']
+        parent = pdcupdater.utils.ensure_release_component_exists(
+            pdc, release_id, name)
+
+        # Go to sleep due to a race condition that is koji's fault.
+        # It publishes a fedmsg message before the task is actually done and
+        # committed to their database.  In the next step, we try to query
+        # them -- but if the task isn't done, we get an exception.
+        time.sleep(1)
+
+        # First, go through all of the relationships that we learn from koji,
+        # and add them to PDC.  Some may already be present, but we may add new
+        # ones here.
+        koji_relationships = self.get_koji_relationships_from_build(
+            self.koji_url, msg['msg']['build_id'])
+        for type, child in koji_relationships:
+            child = pdcupdater.utils.ensure_release_component_exists(
+                pdc, release_id, child)
+            pdcupdater.utils.ensure_release_component_relationship_exists(
+                pdc, parent=parent, child=child, type=type)
+
+        # Lastly, go through all of the relationships that we know of now in
+        # PDC and find any that do not appear in koji.  These must be old
+        # relationships that are no longer relevant.
+        pdc_relationships = list(self._yield_pdc_relationships(pdc, release_id))
+        for entry in pdc_relationships:
+            pdcupdater.utils.delete_release_component_relationship(
```

Comment:
I know in the comments you state that you are filtering based on relationships that are no longer in Koji, but I'm not seeing that here. I must be missing what you're trying to accomplish.